python/desc/imsim/skyModel.py

```diff
@@ -64,7 +64,9 @@ def __init__(self, obs_metadata, seed=None, addNoise=True, addBackground=True):
 
     def addNoiseAndBackground(self, image, bandpass=None, m5=None,
                               FWHMeff=None,
-                              photParams=None):
+                              photParams=None,
+                              tree_ring_center=galsim.PositionD(0, 0),
```

Comment:
@danielsf I'm passing the tree ring info as parameters here, but that's inconsistent with the interface of this method in the base class.   I was thinking that it would make more sense to package the tree ring info with the image data in a detector object and pass that detector object instead of `image`.  This requires more extensive changes in sims_GalSimInterface, however.   What do you think?

For reference, see the [`TreeRings`](https://github.com/rmjarvis/sims_GalSimInterface/blob/ffe3953497b88f1df98d906c689610e17adc8411/python/lsst/sims/GalSimInterface/galSimInterpreter.py#L461) class I added to `sims_GalSimInterface`.

Reply:
I'm not sure how I feel about storing image data in the `detector` object.  I'd originally envisioned `detector` as just a container for parameters pertaining to the physical detector.  I agree that this means that the tree ring parameters should probably live in `detector`.  Could we pass `detector` into `addNoiseAndBackground` alongside `image`?  I imagine that someday there will be even more detector-specific parameters that `addNoiseAndBackground` will need to know about.

Reply:
Sure, that proposal sounds fine.  When I wrote that comment I was conflating the `GalSimInterface.detectors` attribute with the `GalSimInterface.detectorImages` attribute.  So we'd just need to add the tree ring info to the GalSimDetector class and pass in one of those to `addNoiseAndBackground`.